tlatools/org.lamport.tlatools/test/tlc2/util/BufferedRandomAccessFileTest.java

```diff
@@ -170,4 +168,155 @@ public void testInvalidateBufferedData() throws IOException {
24, 104, -19, -87, -9, 121, -43, -83, 44, 84, 89, 39, 61, 60, -113, -46, -26, 56, 113, -121, 60, 90, -46, -70, 37, -95, 10, 18, 89, 21, 103, -77, 30, 10, -20, -23, 109, 81, -65, 7, 29, 96, -60, -88, -38, -65, -108, -127, -79, 106, -28, 63, -3, 31, -93, -74, -31, 18, 66, 37, 13, 56, -99, 2, -59, 29, -86, 33, -75, -76, 75, 61, 50, -64, -21, -122, -29, -70, -66, 8, 93, -28, 88, 40, -92, -10, 0, -11, -92, 40, 87, 40, 5, 35, -87, -86, -38, 22, 104, 113, 83, -70, 69, 32, -56, -27, -84, -78, 43, -125, -74, -84, -60, -46, -37, 99, 61, 86, -106, 56, -115, -41, 29, -28, -82, -111, 98, 17, -54, -64, 52, 2, 125, -118, 23, 65, -30, -125, 119, -102, 91, -71, -108, 75, 7, -73, -121, 58, -55, -81, 31, -85, -96, 76, -84, 66, -119, -29, 57, -126, -122, -69, -10, 124, -98, 97, -81, -23, -89, -97, 14, 12, 109, 79, -24, 38, 8, 43, 89, 108, -68, 23, -114, 112, 70, -123, 76, 72, -84, 10, 25, -46, 4, 8, 121, -59, -43, 98, 18, -114, -32, 48, 43, 26, 49, -33, 39, -68, -18, -75, 39, -21, -127, -64, -28, 74, -53, 119, -40, -62, 64, -54, 3, 44, -68, -92, -52, -80, 116, 58, -121, 108, 123, 82, -94, 106, -53, 80, -121, 16, -91, -80, 52, -106, -15, 68, 91, 125, -33, -53, 66, 124, 4, 69, 17, -127, 43, 58, -111, 70, -127, -112, -126, -86, 10, 3, -15, -125, 51, -47, 107, -70, 91, 58, 2, -82, -32, -62, -114, -92, 22, 35, 78, -45, -56, -37, -104, 15, 83, -41, -10, -68, 56, 75, -65, -10, 69, 117, 79, -21, -76, 24, -8, 63, 5, 61, 100, 27, -111, -59, -53, 104, 104, -15, -76, 59, -13, 91, -84, 86, 57, -66, -118, -57, -122, -95, 44, -100, -124, -16, -83, 38, 109, 1, 107, 23, -40, -127, -3, 80, 60, 0, -69, -58, -79, -27, 67, 77, -2, 125, 57, -108, 102, 64, 33, 1, -112, -3, -101, -4, 69, 65, 61, 0, -45, 109, 74, -77, -48, -43, -102, -126, -113, -40, -6, 41, 50, -30, -88, -73, -9, -77, 108, -99, 103, -85, 110, 49, 42, -81, 88, -99, -84, -71, 100, -69, 45, 127, -108, -119, 44, 62, 30, 42, 42, -63, 111, -68, 14, -123, -8, -90, -1, 71, 47, 122, 77, 69, 33, -17, 97, 110, 19, 75, -2, -63, 71, 69, 23, 28, 113, -14, -17, 60, -20, 85, 34, 100, 85, 28, -50, 64, 44, -29, -52, -54, -38, -83, 46, 89, -114, -89, -14, -83, -53, 93, 37, -7, 60, -51, -1, -58, 29, 16, 97, 37, 13, 9, 126, -99, 90, 34, -34, 4, 27, 59, 40, -46, 123, -128, -94, 83, 68, 101, -48, -67, -96, -48, 47, -27, -40, -101, -85, -60, 13, -81, -115, -128, 116, 9, -67, 69, 82, -128, 13, 118, -82, -103, 112, 7, 62, -19, 70, -17, 19, -57, -94, -67, -76, -76, -49, 51, 110, 113, -78, -110, 120, 50, -64, -127, 32, -43, 79, -52, 66, -56, 17, -73, 86, 122, 22, -122, -37, 19, 61, 113, 98, 121, 66, 95, -117, -127, -37, -118, 37, 47, -2, -83, -48, 46, 16, -117, 3, 27, 23, -94, -34, -38, 1, -63, -2, 79, 31, -118, 41, 44, -86, -11, -42, 43, 63, 121, -114, -24, -52, -24, 14, 16, 80, -87, -36, 72, -45, -63, -23, 0, -16, 115, -81, -45, -39, 75, 3, -40, -91, 110, 28, 4, 55, -52, -76, 109, 92, 105, -94, 28, 65, -107, -50, 65, -68, -109, 108, 77, 35, -77, -52, -77, 36, 53, 126, 72, -6, 70, -18, 64, 86, -4, -36, 1, -111, -28, 116, 19, 15, 71, -12, -15, 124 };
```

Comment:
nit: It seems like we can create this case more easily.

Doesn't the following trigger the original `AssertionError`?

```
raf.seek(9247);
raf.setLength(71);
raf.seek(1675);
raf.write(87);
```